### PR TITLE
Feature/396 data page s3 config file

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-396](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-396)

## Main Changes:
* Moved the content of the **Data** page into a JSON file that can be stored in an S3 bucket.

## Steps To Test:
1. Open http://localhost:3000/data and https://mywaterway.epa.gov/data in side-by-side tabs.
2. Verify that the content, style, glossary items, and links throughout the pages are identical.

### CODE REVIEW

File: app/client/src/components/shared/DataContent.js [hunk 1]
```diff
@@ -113,515 +125,101 @@ function DataContent() {
     };
   }, []);
 
+  const { data, status } = useDataSourcesContext();
+
+  useLayoutEffect(() => {
```
Comment (cschwinderg) [inline app/client/src/components/shared/DataContent.js:130]:
```suggestion
  useEffect(() => {
```

Is there a specific reason you used `useLayoutEffect` here? It seems to work fine with just a plain `useEffect`. 

The react documentation says that when possible to prefer using `useEffect` over `useLayoutEffect`. The docs also say that `useLayoutEffect` can negatively impact performance.

image

File: app/client/src/components/shared/DataContent.js [hunk 2]
```diff
@@ -113,515 +125,101 @@ function DataContent() {
...
+
+  if (status === 'fetching') return <LoadingSpinner />;
+
+  if (status === 'failure') {
+    return (
+      <div css={modifiedErrorBoxStyles}>
+        <p>{dataContentError}</p>
+      </div>
+    );
+  }
```
Comment (cschwinderg) [inline app/client/src/components/shared/DataContent.js:157]: Can you update this to have padding around the errorbox? See the `pageErrorBoxStyles` in the `pages\WaterbodyReport.js` component.

image

File: app/server/app/public/data/data.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
+{
+  "intro": "<em>How’s My Waterway</em> pulls data from multiple databases at EPA and other federal agencies through web services. Below is a summary of data included in <em>How’s My Waterway</em>.",
+  "content": [
+    {
+      "description": "ATTAINS contains information on water quality assessments, impaired waters, and total maximum daily loads (TMDLs), through data submitted by states under Clean Water Act sections <span data-glossary-term data-term='303(d) listed impaired waters (Category 5)'>303(d)</span> and 305(b).",
+      "extraContent": "<h2 style='display:block;margin-bottom:0.25rem;font-size:1.125em;line-height:1.125;font-family:Source Sans Pro,Helvetica Neue,Helvetica,Roboto,Arial,sans-serif;font-weight:bold;padding-bottom:0'>Impairment Category Filtering Tool:</h2><p><a href='attains' target='_blank' rel='noopener noreferrer'>How ATTAINS data are grouped in How’s My Waterway</a></p>",
```
Comment (cschwinderg) [inline app/server/app/public/data/data.json:6]: Production has more padding above `Impairment Category Filtering Tool:`. Can you add more padding there, so it matches production?
image

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well! I like how you used span to dynamically pull in the glossary component. 

I left some comments below, mostly about styles.

File: app/server/app/public/data/data.json [hunk 4]
```diff
@@ -0,0 +1,115 @@
+{
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/server/app/public/data/data.json:1]: Good point, I suppose I was trying to match the tab title (which is indeed generic).

File: app/client/src/components/shared/DataContent.js (hunk 1 above)
Comment (maxdiebold-erg) [reply app/client/src/components/shared/DataContent.js:130]: Yeah I think I had it mixed up. I thought it fired after all elements are rendered (so all the `span`s would be in place), but it looks like it instead concerns layout measurements.

**Review by cschwinderg — APPROVED** [review]
Thanks for making those changes.

**Pull request merged**